packages/ember-views/tests/views/checkbox_test.js

```diff
@@ -139,3 +139,36 @@ QUnit.test('checking the checkbox updates the value', function() {
   equal(!!checkboxComponent.$().prop('checked'), false, 'after clicking a checkbox, the checked property changed');
   equal(get(checkboxComponent, 'checked'), false, 'changing the checkbox causes the view\'s value to get updated');
 });
+
+QUnit.test('sanity check for indeterminate', function() {
+  /* global jQuery */
+  var checkbox = jQuery('<input type="checkbox">').appendTo('#qunit-fixture');
+
+  var changeEventFired = false;
+  checkbox.on('change', function(e) { changeEventFired = true; });
+
+  // The presence of indeterminite: true on the checkbox causes phantomjs to not dispatch the
+  // change event.  Comment this out and re-run this test.
+  checkbox.prop('indeterminate', true);
```

Comment:
We can feature detect, and just not set `indeterminate` if it prevents `change` events.

A similar thing is done in https://github.com/emberjs/ember.js/blob/master/packages/ember-testing/lib/support.js. You can do the detect in https://github.com/emberjs/ember.js/blob/master/packages/ember-views/lib/views/checkbox.js so that it runs at boot time (only once, not once for each instance).